unit-tests/heitech.zer0mqXt.core.tests/RequestReplyTests.cs

```diff
@@ -235,6 +235,27 @@ public void TryRespond_on_same_socket_instance_still_throws()
             Assert.Throws<ZeroMqXtSocketException>(a);
         }
 
+        [Fact]
+        public void Retry_Rq_After_some_seconds_and_register_server_in_meantime_works()
+        {
+            // Arrange
+            Response capturedResponse = null;
+            using var socket = ConfigurationTestData.BuildInProcSocketInstanceForTest("retry-socket", timeoutInMs: 500);
+            // request in background thread
+            Task.Run(async () => capturedResponse = await socket.RequestAsync<Request, Response>(new Request { RequestNumber = 42 }));
+            // wait a second for retry
```

Comment:
Comment seems incorrect.

Reply:
Hm? I dont think so. The request is performed in a task which will return control to the main thread (aka using some other thread, most likely a threadpool thread). So I guess here the comment fits perfectly no?

Reply:
Oh or did you refer to the second for retry and then it just is 250 ms xD? fixed that one.

Reply:
nevertheless, done i guess

Reply:
I was indeed referring to the 1 second vs 250 ms ambiguity.

Reply:
Classy Stephen Cleary: [There Is No Thread](https://blog.stephencleary.com/2013/11/there-is-no-thread.html)
I do dot understand why you're capturing a task in a task.
Why dot directly start the task and await it later like this:
```csharp
// start task
var capturedResponseTask = socket.RequestAsync<Request, Response>(new Request { RequestNumber = 42 });
// Act...
// Wait for task completion
var capturedResponse = await capturedResponseTask;
Assert.NotNull(capturedResponse);
```
Also, what is the purpose of ` Thread.Sleep(100)` here?

Reply:
Ok thanks for Clearyfing ;D 
Still IOC Completion ports use backround threads, this is I/O so I do not totally get what he is arguing :X.

I will implement your suggested changes, and then we can get rid of thread.sleep too 
But we still need Task.Delay instead.

I used the Thread.Sleep for two reasons:
1.) The Test Method returns void.
2.) A delay is required, so the first request without a server leads to the use of the retry policy.

Reply:
> IOC Completion ports use backround threads

They don't :)
A device driver notifies the CPU via an interrupt about the completion of an event.
Yes, there is a I/O thread pool which is subscribed to this handle but only to continue the execution of the task.
However, this only applies to Windows, under Unix I/O is a blocking operation and is indeed deferred to another task for background execution:
https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/IO/FileStream.Unix.cs#L720-L727
Hopefully this will change with one of the next .Net releaes:
https://github.com/dotnet/runtime/issues/12650

`Represents an asynchronous operation` is what the docs say about `
System.Threading.Tasks.Task` and that's exactly what it is. How this is archived is implementation specific.

Reply:
2 month late, but i finally came around to implement it :D